twine/auth.py

```diff
@@ -53,13 +59,41 @@ def username(self) -> Optional[str]:
     @property
     @functools.lru_cache()
     def password(self) -> Optional[str]:
+        if self.oidc:
+            # Trusted publishing (OpenID Connect): get one token from the CI
+            # system, and exchange that for a PyPI token.
+            from id import detect_credential  # type: ignore
```

Comment:
Other than it being an optional dependency, is this an import that would cause a slow-down here?

Reply:
Looking at the code, I don't think there's any particular reason it should be slow. I just did this as a simple way to deal with the optional dependency.

Reply:
So it we import it above and define what we need as None if we can't, we can check instead of allowing an Import error to bubble up to let folks know they need the dependency installed first.